karapuzio/assignment1/q1_anagrams/Anagrams.java

```diff
@@ -0,0 +1,85 @@
+package edu.codeU.assignment1.anagrams;
+
+import java.util.Map;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+
+/**
+ * The class is used for checking if all words in one sentence
+ * have an anagram in another with appropriate number of repeating.
+ */
+public class Anagrams{
+
+    /**
+     * The method is used for dividing sentence to word with or without reversing.
+     * @param sentence is initial string.
+     * @param reverseFlag responsible for the reversing of the string.
+     * @return the map with keys as words and values as count of words.
+     */
+    private static Map<String, Integer> splitSentence2Word(String sentence, boolean reverseFlag){
+        String regex = "[^a-zA-Z']+";
+        String[] split = sentence.split(regex);
+        List<String> tokens = Arrays.asList(split);
+        Map<String, Integer> words = new HashMap<>();
+        for (String token: tokens){
+            String word = token.toLowerCase();
+            if (words.containsKey(word)){
+                int count = words.get(word);
+                words.put(word, count + 1);
+            }
+            else{
+                words.put(word, 1);
+            }
+        }
+        return words;
+    }
+
+    /**
+     * Present word concatenation of chars and number of presentations.
+     * @param words initial Map
+     * @return transforming Map with words
+     */
+    public static Map<String, Integer> representWords(Map<String, Integer> words){
```

Comment:
Interesting solution :) However there is a problem with it: it limits us very much to only use letters. What if at some point we would like to support for numbers as well? It would start misbehaving. I would suggest just putting the same latter in the string as many times as it appears in the original word (so basically sorting the word) or maybe escaping the characters in the newWord string so that there is a distinction between a number meaning a number and a number being a character in the original word.

Reply:
I got you. :) Or I can also split the character and its amount with a special character, for example, the space. Because in words there can;t be a gap. :) Is it ok?

Reply:
Can you give an example of how that would look? How would the output look for "mama" and "mama333"?